src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -591,6 +819,47 @@ private static void InternalWriteAllLines(TextWriter writer, IEnumerable<String>
             }
         }
 
+        public static Task WriteAllLinesAsync(
+            string path, IEnumerable<string> contents, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return WriteAllLinesAsync(path, contents, UTF8NoBOM, cancellationToken);
+        }
+
+        public static Task WriteAllLinesAsync(
+            string path, IEnumerable<string> contents, Encoding encoding,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (path == null)
+                throw new ArgumentNullException(nameof(path));
+            if (contents == null)
+                throw new ArgumentNullException(nameof(contents));
+            if (encoding == null)
+                throw new ArgumentNullException(nameof(encoding));
+            if (path.Length == 0)
+                throw new ArgumentException(SR.Argument_EmptyPath, nameof(path));
+
+            Contract.EndContractBlock();
+
+            return cancellationToken.IsCancellationRequested
+                ? Task.FromCanceled(cancellationToken)
+                : InternalWriteAllLinesAsync(AsyncStreamWriter(path, false, encoding), contents, cancellationToken);
+        }
+
+        private static async Task InternalWriteAllLinesAsync(TextWriter writer, IEnumerable<String> contents, CancellationToken cancel)
+        {
+            Debug.Assert(writer != null);
+            Debug.Assert(contents != null);
+
+            using (writer)
+            {
+                foreach (String line in contents)
+                {
+                    cancel.ThrowIfCancellationRequested();
+                    await (line == null ? writer.WriteLineAsync() : writer.WriteLineAsync(line)).ConfigureAwait(false);
```

Comment:
AFAICT this null check being necessary is a bug in TextWriter. Will investigate, report, and possibly fix, tomorrow.